### ISSUE TITLE
[Bug]: Bat mode problem (modcompat problem)

### ISSUE THREAD

**Issue by TakehisaYuji** [TeamLapen/Vampirism#1456]
### Minecraft Version

1.20.1

### (Neo)Forge Version

47.3.0

### Vampirism Version

1.10.11

### Modpack

[modpack.zip](https://github.com/user-attachments/files/18006141/modpack.zip)

### Singleplayer/Multiplayer

Both

### What happened?

So i posted a bug report here some time ago (https://github.com/TeamLapen/Vampirism/issues/1397), the issue was that the bat mode was "randomly" turning off, then after i updated a few mods the problem seemed to go away, but as it turns out, nothing changed.
i thought the problem was with the mobs from other mods or maybe the entities or something like that (i tried everything)

but it turns out, that my actions is what caused the problem to occur.

So. the problem is that when i am in a bat mode, scrolling the hotbar and selecting any item or just scrolling from one to another, causes the flight to stop and me falling as a result, and also taking the fall damage and still being a bat at the same time.
the same thing happens when i'm trying to drop items while in a bat mode.

### Reproduce steps

1. fly in a bat mode.
2. select an item in the hotbar / drop an item.

i already tried disabling all the vampirism addons that i had and tried recreating this with just a vampirism mod without any other ones, no luck.

**Comment by TakehisaYuji** [comment]
I found the problem,

the armorplus mod is the reason why all of this happens (or maybe its the libraries that come with it, like the geckolib or dragonlib)

well anyway, i dont know who i should tell this to, maybe the armorplus devs or you guys, i guess i will just leave it here for now and wait for a response

**Comment by maxanier** [comment]
See https://github.com/sokratis12GR/ArmorPlus/issues/281

@TakehisaYuji Thank you for narrowing down this tricky problem!

@Cheaterpaul 
TLDR: On 1.21 this has to be fixed on ArmorPlus side.
On 1.20 probably as well, but we could consider to force the `mayFly` attribute in `VampireBatAction#onUpdate' each tick.
(maybe also only if a config option is set)

**Comment by maxanier** [comment]
Unfortunately, ArmorPlus does not seem to receive any updates.

In the next 1.20 Vampirism update, we are now forcing the `mayFly` attribute to be true.
This does not really solve the problem, because flight is still canceled as soon as you scroll in your taskbar.
However, at least you can immediately start flying again without having to toggle bat mode.

There is nothing else we can reasonably do, sorry.
